cornell-dti/zing-lsc
# Notes frontend

# Summary

This PR implements the front-end side of the student notes feature. With this PR, the notes feature should be finished. 

- [x] note modals for each student
- [x] alerts for note saves + save failures

### Current Implementation of Edit + Checkbox icon. 
![image](https://user-images.githubusercontent.com/46132945/178858686-c161df76-6402-4da4-8168-45a0f19d5143.png)

### Notes
An appropriate snack bar alert appears when the note is successfully saved and if a save error occurs. 

## Test Plan 

Optional: `npm run populate` + send own survey in. 
Output: Multiple students in multiple groups for multiple classes have their correct notes saved and updated. 


2. Create a note in another class (PHYS 2213) 
<img width="1162" alt="image" src="https://user-images.githubusercontent.com/46132945/177435940-1a41aa77-d19f-4063-ad42-b4866db605ac.png">

3. Create a note for another class (MATH 1920) but in an unmatched group
<img width="1164" alt="image" src="https://user-images.githubusercontent.com/46132945/177435993-802a1f31-4cd1-432d-8f1f-5fa70cbc411b.png">

4. Check DB + reload to see if note is saved for each class 
<img width="620" alt="image" src="https://user-images.githubusercontent.com/46132945/177436018-0399d798-134c-4c9f-8866-069adb3c5080.png">

And yes, notes are still saved. 

5. Check if updated notes are saved 
<img width="1129" alt="image" src="https://user-images.githubusercontent.com/46132945/177436164-7446176c-0eb2-4906-8a69-a8de3a578f9c.png">
6. Saved in DB:
<img width="615" alt="image" src="https://user-images.githubusercontent.com/46132945/177436212-33bbeda3-575a-45e8-83cf-4d3981f92a74.png">
